app/collections/tweets_by_week_collection.rb

```diff
@@ -0,0 +1,20 @@
+class TweetsByWeekCollection < Patterns::Collection
+  private
+
+  def collection
+    group_by_weeks
+  end
+
+  def group_by_weeks
+    weeks.map do |week|
+      tweets = subject.where(tweet_created_at: week)
+      Tweets::Week.new(week, tweets)
+    end
+  end
+
+  def weeks
+    subject.order(:tweet_created_at).map do |tweet|
+      tweet.tweet_created_at.beginning_of_week..tweet.tweet_created_at.end_of_week
```

Comment:
simplify this, so that we do not go over all tweets to get range
think in a different way, that we specify time range, then based on that, we create weeks, days, then we get tweets for a day